homeassistant/components/mvglive/sensor.py

```diff
@@ -147,75 +132,89 @@ def update(self) -> None:
-        self._include_tram = "Tram" in self._products
-        self._include_bus = "Bus" in self._products
-        self._include_sbahn = "S-Bahn" in self._products
-        self.mvg = MVGLive.MVGLive()
+        self._station = MvgApi.station(self._station_name)
+        if self._station:
+            self.mvg = MvgApi(self._station["id"])
         self.departures = []
 
     def update(self):
         """Update the connection data."""
+        if not self._station or not self.mvg:
+            self.departures = []
+            _LOGGER.warning(
+                "Station cannot be found by name: change name or use station id, e.g. de:99232:2353"
+            )
+            return
         try:
-            _departures = self.mvg.getlivedata(
-                station=self._station,
-                timeoffset=self._timeoffset,
-                ubahn=self._include_ubahn,
-                tram=self._include_tram,
-                bus=self._include_bus,
-                sbahn=self._include_sbahn,
+            _departures = self.mvg.departures(
+                offset=self._timeoffset,
+                limit=self._number,
+                transport_types=[
+                    transport_type
+                    for transport_type in TransportType
+                    if transport_type.value[0] in self._products
+                ]
+                if self._products
+                else None,
             )
         except ValueError:
             self.departures = []
             _LOGGER.warning("Returned data not understood")
             return
         self.departures = []
-        for i, _departure in enumerate(_departures):
+        for _i, _departure in enumerate(_departures):
             # find the first departure meeting the criteria
             if (
                 "" not in self._destinations[:1]
                 and _departure["destination"] not in self._destinations
             ):
                 continue
 
-            if (
-                "" not in self._directions[:1]
-                and _departure["direction"] not in self._directions
-            ):
+            if "" not in self._lines[:1] and _departure["line"] not in self._lines:
                 continue
 
-            if "" not in self._lines[:1] and _departure["linename"] not in self._lines:
-                continue
+            time_to_departure = _get_minutes_until_departure(_departure["time"])
 
-            if _departure["time"] < self._timeoffset:
+            if time_to_departure < self._timeoffset:
                 continue
 
             # now select the relevant data
             _nextdep = {}
-            for k in ("destination", "linename", "time", "direction", "product"):
+            for k in ("destination", "line", "type", "cancelled", "icon"):
                 _nextdep[k] = _departure.get(k, "")
-            _nextdep["time"] = int(_nextdep["time"])
+            _nextdep["time_in_mins"] = time_to_departure
```

Comment:
I think @joostlek wants to say that maybe it is better to just leave only changes that are essential for making it work as before?